src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,122 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DoneCommand;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.GetCommand;
+import duke.command.ListCommand;
+import duke.command.UpdateCommand;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Makes sense of the user command.
+ */
+public class Parser {
+    /**
+     * Parses the specified command and returns the corresponding Command object.
+     *
+     * @param command The command to be parsed.
+     * @return The corresponding Command object.
+     * @throws DukeException If the command is not supported.
+     */
+    public static Command parse(String command) throws DukeException {
+        Command cmd;
+        String[] commandArray = command.split(" ", 2);
+        switch (commandArray[0]) {
+        case "todo":
+            try {
+                cmd = new AddCommand(commandArray[0], commandArray[1]);
+            } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
You can have your own meaningful exceptions in future implementations e.g. MissingArgumentException or InvalidArgumentException